ext/snmp/snmp.c

```diff
@@ -626,6 +626,31 @@ static void php_snmp_internal(INTERNAL_FUNCTION_PARAMETERS, int st,
 }
 /* }}} */
 
+static void php_snmp_zend_string_release_from_char_pointer(char *ptr) {
+	if (ptr) {
+		zend_string *pptr = (zend_string *)(ptr - XtOffsetOf(zend_string, val));
+		zend_string_release(pptr);
+	}
+}
+
+static void php_free_objid_query(struct objid_query *objid_query, HashTable* oid_ht, const zend_string *value_str, const HashTable *value_ht, int st) {
+	if (oid_ht) {
+		uint32_t count = zend_hash_num_elements(oid_ht);
+
+		for (uint32_t i = 0; i < count; i ++) {
+			snmpobjarg *arg = &objid_query->vars[i];
+			if (!arg->oid) {
+				break;
+			}
+			if (value_ht && !value_str) {
```

Comment:
I think you only need to check value_ht, and then you can remove the value_str argument.

Reply:
I ll check a bit later, it s mainly [because this](https://github.com/php/php-src/pull/16969/files#diff-a49eb43323a4f7bf6ea3611b3e427163bfcf29694778ad35e9ee4d9e81241bbdL721) but we ll see.

Reply:
Yes I see, but note that the way the arguments are parsed only value_ht or value_str can be non-NULL:
https://github.com/php/php-src/blob/2c46d83c80e1c20730348bc2beb3d720c057786a/ext/snmp/snmp.c#L1202